Lib/unittest/case.py

```diff
@@ -91,12 +90,13 @@ def _addSkip(result, test_case, reason):
                       RuntimeWarning, 2)
         result.addSuccess(test_case)
 
-def _addError(result, test, exc_info):
-    if result is not None and exc_info is not None:
-        if issubclass(exc_info[0], test.failureException):
-            result.addFailure(test, exc_info)
+def _addError(result, test, exc):
+    assert not isinstance(exc, tuple)
+    if result is not None and exc is not None:
+        if isinstance(exc, test.failureException):
+            result.addFailure(test, exc)
         else:
-            result.addError(test, exc_info)
+            result.addError(test, exc)
```

Comment:
Maybe I should revert this change, in case there are use-defined subclasses of TestResult that expect the tuple?

Reply:
Yeah, I think the folks still using unittest are often in interesting and subtle places (Twisted's trial springs to mind...) so I think we really want to make sure 100% backwards compatibility (even in internal APIs *sigh*) is preserved if at all possible.

I'm curious: where are you finding yourself calling `addError` and friends in your code?

Reply:
I'm not calling addError in my code. We're trying to remove the (typ, exc, tb) triplet from the language. The interpreter doesn't need it anymore, it only exists for backwards compatibility.

Reply:
I made this a draft PR, it need more thought. If we can't make unittest use the new APIs then I'm not sure there is a point committing this. We would need a plan for how to migrate off the triplet. 

This is related to the PEP707 discussion, so we could wait to see what comes out of that.